vertx-web/src/test/java/io/vertx/ext/web/templ/TemplateTest.java

```diff
@@ -72,6 +72,31 @@ private void testRelativeToRoutePath(String pathPrefix) throws Exception {
   }
 
   @Test
+  public void testTemplateHandlerSingleRoute() throws Exception {
+    TemplateEngine engine = new TestEngine(false);
+    router.route("/test-template")
+             .handler(context -> {
+               context.put("foo", "badger");
+               context.put("bar", "fox");
+               context.next();
+             })
+            .handler(TemplateHandler.create(engine, "somedir", "text/html"));
```

Comment:
The template handler when used this way expects a directory as base path that is how it matches the files later on, if you want to use a single file you can call the render method directly on your handler